ports/atmel-samd/Makefile

```diff
@@ -99,7 +99,7 @@ endif
 
 #Debugging/Optimization
 ifeq ($(DEBUG), 1)
-  CFLAGS += -ggdb
+  CFLAGS += -ggdb -Og
```

Comment:
What is -Og? Do SAMD21 builds still fit?

Reply:
-Og is "optimizations that don't impede debugging".  We had added it on nrf awhile back.

Actually, this makes the build with DEBUG=1 BOARD=trinket_m0 fail, but not due to size.  
```
../../py/vm.c: In function 'mp_execute_bytecode':
../../supervisor/shared/translate.c:98:66: error: inlining failed in call to always_inline 'translate': optimization level attribute mismatch
 inline __attribute__((always_inline)) const compressed_string_t* translate(const char* original) {
                                                                  ^
../../py/vm.c:255:44: note: called from here
                             mp_obj_t obj = mp_obj_new_exception_msg(&mp_type_NameError, translate("local variable referenced before assignment"));
                                            ^
```
I'll rebase this out.